src/operator/spatial_transformer.cc

```diff
@@ -28,7 +28,7 @@
 namespace mshadow {
 template<typename DType>
 bool between(DType value, int lowerBound, int upperBound) {
-  return (value >= lowerBound && value <= upperBound);
+  return (static_cast<int>(value) >= lowerBound && static_cast<int>(value) <= upperBound);
```

Comment:
as I mentioned to @rahul003 before this is problematic

Reply:
This whole function is problematic IMHO, but as the prototype stands, this shouldn't change the comparison which is occurring, since the compiler will generate roughly the same comparison (comparing mantissa only).  It's causes compile error so it needs to be dealt with.

Reply:
Also note that since the comparisons are >= and <=, this correctly handles the rounding

Reply:
I think the proper fix is to change DType to just int if it's never used for float.

Otherwise your fix changes behavior. Suppose value = 5.5 and upperbound = 5

Reply:
done